v2/samples/containerservice/v1api20231102preview/v1api20231102preview_managedcluster.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: containerservice.azure.com/v1api20231102preview
+kind: ManagedCluster
+metadata:
+  name: sample-managedcluster-20231102preview
+  namespace: default
+spec:
+  location: westus3
+  owner:
+    name: aso-sample-rg
+  dnsPrefix: aso
+  agentPoolProfiles:
```

Comment:
Just confirming, `agentPoolProfiles` looks like an embedded `ManagedClustersAgentPool`. 
Do we need to merge these as how we do for other resources (vnet, loadbalancer, etc)?

Reply:
I don't understand the point you're making. This is done the same way as existing versions of `ManagedCluster`, which I assume are correct.

Reply:
Need to try out the steps below: 
1. Create MC with agentPoolProfiles
2. Create ManagedClustersAgentPool resource
3. Update `agentPoolProfiles` to add another `AgentPool`

In above, we need to make sure, the ManagedCluster resource does not override the AgentPools and remove the `ManagedClustersAgentPool` we created in step 2

Reply:
Ok nah, `ManagedCluster` api is smart enough to handle that by itself: 

>  RESPONSE 400: 400 Bad Request                                                                                                                                                              
>  ERROR CODE: BadRequest                                                                                                                                                                      --------------------------------------------------------------------------------                                                                                                            "message": "A new agent pool was introduced. Adding agent pools to an existing cluster is not allowed through managed cluster operations. For agent pool specific change, please use per agent pool operations: https://aka.ms/agent-pool-rest-api", "subcode": ""